### PR TITLE
update CMAKE_BUILD_TYPE policy

### PR DESCRIPTION
**Pull request by asartori86**
update required after https://github.com/dealii/dealii/pull/3088

### CODE REVIEW

**Comment by asartori86** [comment]
@luca-heltai could you please check what is wrong with docker for the debug version?

I cannot understand the issue. Thanks

**Comment by luca-heltai** [comment]
I think the issue is related to the fact that the latest docker images are broken. I'm recompiling the whole thing right now...

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,7 +69,21 @@ FIND_PACKAGE(deal.II 8.4.0 REQUIRED
...
+
+IF(DEFINED CMAKE_BUILD_TYPE)
+   SET(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE} CACHE STRING 
+       "Choose the type of build, options are: Debug, Release or DebugRelease"
+       )
+ELSE()
+   SET(CMAKE_BUILD_TYPE ${DEAL_II_BUILD_TYPE} CACHE STRING
+       "Choose the type of build, options are: Debug, Release or DebugRelease"
+       )
+ENDIF()
```
Comment (tamiko) [inline CMakeLists.txt]: There is no need for the `IF`-`ELSE` block here. You can simply set unconditionally

```
SET(CMAKE_BUILD_TYPE ${DEAL_II_BUILD_TYPE} CACHE STRING 
  "Choose the type of build, options are: Debug, Release or DebugRelease"
  )
```

If the variable `CMAKE_BUILD_TYPE` is already set in cache, the `SET(... CACHE ...)` function will not override it.
(If no additional `FORCE` is supplied as keyword.)

**Review by tamiko — CHANGES_REQUESTED** [review]
Only the `SET` statement is necessary, no need for the `IF`-`ELSE` block.

File: CMakeLists.txt (hunk 1 above)
Comment (asartori86) [reply CMakeLists.txt]: Oh yes! thanks indeed @tamiko!

**Comment by asartori86** [comment]
to me this is ready to be merged. all the tests work fine (except those referenced in #296)

**Comment by asartori86** [comment]
I merge this and open an issue that travis is broken

**Pull request merged**